redhat-appstudio/service-provider-integration-operator
# Enable kube-linter and resolve problems reported by it

### What does this PR do?
 - Add required security options
 - Exclude vault-test pod which cannot be configured properly

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

https://issues.redhat.com/browse/SVPI-314

### How to test this PR?


```
quay.io/redhat-appstudio/pull-request-builds:spi-operator-#
quay.io/redhat-appstudio/pull-request-builds:spi-oauth-#
```
- Verify that kube-lint CI task is passing without errors
- Verify there is no functional degradation
